neo/Network/RPC/NeoAPI.cs

```diff
@@ -0,0 +1,152 @@
+                .Stack.Single().ToStackItem().GetBigInteger();
+            return ((decimal)balance) / (long)NativeContract.GAS.Factor;
+        }
+
+        /// <summary>
+        /// Get Neo Balance
+        /// </summary>
+        /// <param name="addressOrHash">address or hash</param>
+        /// <returns></returns>
+        public uint GetNeoBalance(string addressOrHash)
+        {
+            BigInteger balance = GetTokenBalance(NativeContract.NEO.Hash.ToString(), addressOrHash);
+            return (uint)balance;
+        }
+
+        /// <summary>
+        /// Get Gas Balance
+        /// </summary>
+        /// <param name="addressOrHash">address or hash</param>
+        /// <returns></returns>
+        public decimal GetGasBalance(string addressOrHash)
+        {
+            BigInteger balance = GetTokenBalance(NativeContract.GAS.Hash.ToString(), addressOrHash);
+            return ((decimal)balance) / (long)NativeContract.GAS.Factor;
+        }
+
+        /// <summary>
+        /// Get token balance
+        /// </summary>
+        /// <param name="tokenHash">token script hash</param>
+        /// <param name="addressOrHash">address or hash</param>
+        /// <returns></returns>
+        public BigInteger GetTokenBalance(string tokenHash, string addressOrHash)
+        {
+            UInt160 scriptHash = tokenHash.ToUInt160();
+            UInt160 account = addressOrHash.ToUInt160();
+            return Nep5API.BalanceOf(scriptHash, account);
+        }
+
+        /// <summary>
+        /// The GAS is claimed when doing NEO transfer
+        /// This function will transfer NEO balance from account to itself
+        /// </summary>
+        /// <param name="key">wif or private key</param>
+        /// <returns>The transaction sended</returns>
+        public Transaction ClaimGas(string key)
+        {
+            KeyPair keyPair = key.ToKeyPair();
+            UInt160 toHash = keyPair.ToScriptHash();
+            BigInteger balance = Nep5API.BalanceOf(NativeContract.NEO.Hash, toHash);
+            Transaction transaction = Nep5API.CreateTransferTx(NativeContract.NEO.Hash, keyPair, toHash, balance);
+            RpcClient.SendRawTransaction(transaction);
+            return transaction;
+        }
+
+        /// <summary>
+        /// Transfer NEP5 token balance, with common data types
+        /// </summary>
+        /// <param name="tokenHash">nep5 token script hash</param>
+        /// <param name="fromKey">wif or private key</param>
+        /// <param name="toAddress">address or account script hash</param>
+        /// <param name="amount">token amount</param>
+        /// <param name="networkFee">netwotk fee, set to be 0 will auto calculate the least fee</param>
+        /// <returns></returns>
+        public Transaction Transfer(string tokenHash, string fromKey, string toAddress, decimal amount, decimal networkFee = 0)
```

Comment:
I prefer to use the real types, why strings?

Reply:
Hi Shargon
The string and decimal types are friendly for developers without any block-chain technical experience. And users usually get string type values from CLI and GUI, so I use only string type in NeoAPI functions.

Reply:
Then we should provide both, and in comments add the expected format when is a string